contract/models/contract.py

```diff
@@ -546,8 +546,13 @@ def cron_recurring_create_invoice(self, date_ref=None):
         if not date_ref:
             date_ref = fields.Date.context_today(self)
         domain = self._get_contracts_to_invoice_domain(date_ref)
-        contracts_to_invoice = self.search(domain)
-        return contracts_to_invoice._recurring_create_invoice(date_ref)
+        # Invoice by companies, so assignation emails get correct context
+        companies_to_invoice = self.read_group(domain, ["company_id"], ["company_id"])
+        for row in companies_to_invoice:
+            contracts_to_invoice = self.search(row["__domain"]).with_context(
+                allowed_company_ids=[row["company_id"][0]]
+            )
+            contracts_to_invoice._recurring_create_invoice(date_ref)
```

Comment:
please keep the return

Reply:
invoices records should be then stacked and return at the end.

Reply:
yes that what I mean :)

Reply:
I saw it's used nowhere, but OK no problem on this.